### ISSUE TITLE
login_required after follow any user

### ISSUE THREAD

**Issue by bariscck** [sbertix/Swiftagram#18]
Hi @sbertix. I'm trying to follow user like this;

```
func follow(for userId: String, completion: @escaping (Bool) -> Void) -> Task {
        let task = Endpoint.Friendship.follow(userId)
            .authenticating(with: secret)
            .task { (result) in
                guard let response = try? result.get() else { return }
                print(response)
                completion(true)
            }
        
        task.resume()
        return task
    }
```

but its logged out me and response received like this;

```
{
  "logout_reason" : 3,
  "message" : "login_required",
  "status" : "fail",
  "error_title" : "You've Been Logged Out",
  "error_body" : "Please log back in."
}
```

**Comment by sbertix** [comment]
Yep, it seems to have to do with request changes for accounts marked as `spammer`.  
I can seem to make it work with all accounts I tried, but it's a known issue, apparently.  
I'm definitely investigating this further.  
You're on `2.1.2`, right?

**Comment by bariscck** [comment]
You mean `1.2.1`, yes. I just saw https://github.com/TheM4hd1/SwiftyInsta/issues/173 this, its look like related.

**Comment by sbertix** [comment]
It seems to be case, indeed.  
Unfortunately it's super hard for me to fix an issue I can't replicate.  
I'm gonna try some things later on the `development` branch if you don't mind testing them to see if they solve the issue for you @bariscck

**Comment by sbertix** [comment]
The tentative fix on the `development` branch is now live. It still works for me, so at least I haven't made things work hahahaha
Would you mind letting me know whether it fixes the issue for you or not? 😊 @bariscck

**Comment by bariscck** [comment]
@sbertix I tried but result is same, logged out me. I may have found something, I will try and inform you.

**Comment by sbertix** [comment]
I'm definitely trying to update `Secret` accordingly and push a commit by tomorrow.  
Is this the minimum set of cookies required? Meaning, have you tried removing one at a time until it doesn't work anymore?

**Comment by bariscck** [comment]
@sbertix I tried now, interesting part is its not associate with `Cookie` format, it was false alarm. Im using **Charles Proxy** for monitoring network requests and I copy `Cookie` from another published app and paste it to **Postman** with our format and it works but our `Cookie` always logged out me. 🤔

**Comment by sbertix** [comment]
Just as I worried: this is related to the authentication error I wrote about in **SwiftyInsta**, and there's not much I can do.  
Have you obtained your `Secret` through `BasicAuthenticator` or `WebViewAuthenticator`?

**Comment by bariscck** [comment]
Im using `WebViewAuthenticator`, but this other apps also using `WebView` its interesting, I will check your wroted issue.

**Comment by sbertix** [comment]
It's related to custom user agents and `spammer` accounts.  
[Here](https://github.com/TheM4hd1/SwiftyInsta/issues/163#issuecomment-609105398) my explanation in the original issue.

**Comment by bariscck** [comment]
I switched `WebViewAuthenticator` to `BasicAuthenticator` and its solved this problem. I'll probably make a fake instagram login page.

Thanks @sbertix for your time and helps 😊

**Comment by sbertix** [comment]
I'm still going to try and find a proper fix for `WebViewAuthenticator`, as well as adding a **SwiftagramUI** library for (at least) `UIKit` and `SwiftUI` out-of-the-box login screens, to use with `BasicAuthenticator`.

**Comment by bariscck** [comment]
I think the same thing. But an important recommendation; this should be `HTML` and opened with `WebView`. Otherwise, it probably will be rejected from AppStore review.

**Comment by sbertix** [comment]
I'm more interested in the "proof of concept" thing, rather than the "bypassing the App Store review process restrictions".  
Both implementations present their challenges and downsides (using basic authentication and mimicking a login page are both big no-nos), and app using **Swiftagram** or any other similar library are not safe for the App Store by definition, regardless of the authentication method.  
Most apps relying on this might be naively approved, at first. But they are definitely not gonna stand the test of time.

**tl;dr**: the implementation will not try to mimic existing behavior or "fool" the final user or reviewers. I don't want to be responsible for anyone abusing this functionality, tbh.

**Comment by sbertix** [comment]
I've made a small change to `WebViewAuthenticator` to use the current `Device`, thinking maybe it might be related to Instagram keeping tap on that. Idk.
Would you mind trying the commit on `development` by any chance? @bariscck

**Comment by bariscck** [comment]
@sbertix I tried now but unfortunately result is same.

**Comment by ansh** [comment]
@sbertix Any ideas about this issue?

**Comment by ansh** [comment]
@bariscck Did you ever manage to fix this?
